packages/shared/src/SliderLabels.js

```diff
@@ -0,0 +1,190 @@
+// @flow
+
+import * as React from 'react';
+import { defaultTokens } from '@kiwicom/mobile-orbit';
+import { View } from 'react-native';
+import type { OnLayout } from '@kiwicom/mobile-shared';
+import { type TranslationType } from '@kiwicom/mobile-localization';
+
+import Text from './Text';
+import StyleSheet from './PlatformStyleSheet';
+
+type Props = {|
+  +startLabel: TranslationType,
+  +startValue: number,
+  +endLabel?: TranslationType,
+  +endValue?: number,
+  +max: number,
+  +min: number,
+|};
+
+type State = {|
+  width: number,
+  labelStartWidth: number,
+  labelStartAtMax: boolean,
+  labelEndWidth: number,
+  paddingLeft: number,
+  paddingRight: number,
+|};
+
+export default class SliderLabels extends React.Component<Props, State> {
```

Comment:
Also there is a lot of logic in this class, could you please also add some tests?

Reply:
I added a test